### PR TITLE
<Crusade> Claim NFT drop to v5

### PR DESCRIPTION
**Pull request by kien-ngo**
## Problem solved

Short description of the bug fixed or feature added

---

## PR-Codex overview
The focus of this PR is to refactor NFT claim functionality by updating contract handling and adding transaction confirmation.

### Detailed summary
- Refactored `NFTClaimButton` and `NFTClaimForm` to use `contractAddress` and `chainId`
- Added transaction confirmation for NFT claiming
- Updated contract handling in `NFTClaimForm`

> ✨ Ask PR-Codex anything about this PR by commenting with `/codex {your question}`

### CODE REVIEW

File: apps/dashboard/src/contract-ui/tabs/nfts/components/claim-form.tsx [hunk 1]
```diff
@@ -79,29 +88,32 @@ export const NFTClaimForm: React.FC<NFTClaimFormProps> = ({ contract }) => {
...
-                    label: "error",
-                    error,
-                  });
-                  onError(error);
-                },
+
+            const transaction = claimTo({
+              contract,
+              to: d.to ?? "",
+              quantity: BigInt(d.amount),
```
Comment (MananTank) [inline apps/dashboard/src/contract-ui/tabs/nfts/components/claim-form.tsx:95]: V4 was able to take no `to` field. not sure what's changed in v5

Comment (MananTank) [reply apps/dashboard/src/contract-ui/tabs/nfts/components/claim-form.tsx:95]: @jnsdls can probably verify this one

Comment (kien-ngo) [reply apps/dashboard/src/contract-ui/tabs/nfts/components/claim-form.tsx:95]: it's an extension so ig it is what it is (we have to specify the field)

Comment (jnsdls) [reply apps/dashboard/src/contract-ui/tabs/nfts/components/claim-form.tsx:95]: `claimTo` _has_ to have an address, the behavior in v4 for claim without address would claim to the connected address so we have to do the same here, empty string will throw an error during the mutation here I believe

Comment (jnsdls) [reply apps/dashboard/src/contract-ui/tabs/nfts/components/claim-form.tsx:95]: unless we handle this in the extension (we shouldn't!) the consumer is responsible to pass the address here (meaning dashboard has to pass it) `d.to` is no longer optional here

File: apps/dashboard/src/contract-ui/tabs/nfts/components/claim-form.tsx [hunk 2]
```diff
@@ -23,12 +24,20 @@ import {
...
 }
 
-export const NFTClaimForm: React.FC<NFTClaimFormProps> = ({ contract }) => {
+export const NFTClaimForm: React.FC<NFTClaimFormProps> = ({
+  contractAddress,
+  chainId,
+}) => {
+  const contract = getContract({
+    address: contractAddress,
+    chain: defineChain(chainId),
```
Comment (jnsdls) [inline apps/dashboard/src/contract-ui/tabs/nfts/components/claim-form.tsx:37]: v4tov5definechain util thing here instead?

File: apps/dashboard/src/contract-ui/tabs/nfts/components/claim-button.tsx [hunk 3]
```diff
@@ -40,7 +24,7 @@ export const NFTClaimButton: React.FC<NFTClaimButtonProps> = ({
         onClose={onClose}
         isOpen={isOpen}
       >
-        <NFTClaimForm contract={contractQuery.contract} />
+        <NFTClaimForm contractAddress={contractAddress} chainId={chainId} />
```
Comment (jnsdls) [inline apps/dashboard/src/contract-ui/tabs/nfts/components/claim-button.tsx:27]: I wonder if this implicitly means we are removing support for some older claim types? does v5 claimTo extension handle all the v4 supported cases? I legitimately do not know cc @joaquim-verges @gregfromstl?

Comment (jnsdls) [reply apps/dashboard/src/contract-ui/tabs/nfts/components/claim-button.tsx:27]: if it does mean that then same as with marketplace that's probably OK we just have to make sure support is informed of this - so we're not surprised of _possible_ issues

File: apps/dashboard/src/contract-ui/tabs/nfts/page.tsx [hunk 4]
```diff
@@ -48,6 +49,18 @@ export const ContractNFTPage: React.FC<NftOverviewPageProps> = ({
     "ERC721ClaimCustom",
   ]);
 
+  const detectedClaimState = extensionDetectedState({
```
Comment (jnsdls) [inline apps/dashboard/src/contract-ui/tabs/nfts/page.tsx:52]: nice, I like moving this up - we also have to figure out how to do this same thing in v5 soon

**Review by jnsdls — CHANGES_REQUESTED** [review]

File: apps/dashboard/src/contract-ui/tabs/nfts/components/claim-form.tsx (hunk 1 above)
Comment (kien-ngo) [reply apps/dashboard/src/contract-ui/tabs/nfts/components/claim-form.tsx:95]: correct. just read the code for getClaimParams. I believe if we dont pass it we will get an "execution reverted"

File: apps/dashboard/src/contract-ui/tabs/nfts/components/claim-form.tsx (hunk 2 above)
Comment (kien-ngo) [reply apps/dashboard/src/contract-ui/tabs/nfts/components/claim-form.tsx:37]: yesser

File: apps/dashboard/src/contract-ui/tabs/nfts/components/claim-form.tsx (hunk 1 above)
Comment (jnsdls) [reply apps/dashboard/src/contract-ui/tabs/nfts/components/claim-form.tsx:95]: yep, so form should be invalid here if no address is filled out meaning we should never need an  empty string for this case

Comment (kien-ngo) [reply apps/dashboard/src/contract-ui/tabs/nfts/components/claim-form.tsx:95]: let me check. i think typescript was screaming at me (irrc it didn't understand the required* field from react-form-hook which should prevent the falsy case)

Comment (jnsdls) [reply apps/dashboard/src/contract-ui/tabs/nfts/components/claim-form.tsx:95]: if that's the case just `if(!d.to) {don't try to do the tx just show the error immediately here}`

**Review by jnsdls — APPROVED** [review]
please make sure support is informed that we changed this and to report any uptick in issues with claiming they may see

File: apps/dashboard/src/contract-ui/tabs/nfts/components/claim-button.tsx (hunk 3 above)
Comment (kien-ngo) [reply apps/dashboard/src/contract-ui/tabs/nfts/components/claim-button.tsx:27]: Will imform the guys about this change!

**Pull request merged**